### PR TITLE
#29: [New Rule] Exceptions must not be handled in the same function

### PR DESCRIPTION
**Pull request by larsroettig**
see #29

### CODE REVIEW

File: Magento/ruleset.xml [hunk 1]
```diff
@@ -203,10 +206,6 @@
         <severity>6</severity>
         <type>warning</type>
     </rule>
-    <rule ref="Squiz.Operators.ValidLogicalOperators">
```
Comment (lenaorobei) [inline Magento/ruleset.xml]: Please pull latest changes from `develop`.

File: Magento/ruleset.xml [hunk 2]
```diff
@@ -72,6 +72,9 @@
     <rule ref="Magento.Exceptions.Namespace">
         <severity>8</severity>
     </rule>
+    <rule ref="Magento.Functions.ThrowCatch">
```
Comment (lenaorobei) [inline Magento/ruleset.xml]: Could you please add `<type>warning</type>`. We decided to be as explicit as possible in the ruleset :)

**Comment by larsroettig** [comment]
@lenaorobei fixes for review done

**Comment by lenaorobei** [comment]
Closes #magento/magento-coding-standard#29

**Comment by danmooney2** [comment]
Exceptions that throw with only one `catch` block pass.  
This one fails correctly:
```
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

namespace Magento;

/**
 * Doer
 */
class Doer
{
    /**
     * Do Something
     */
    public function doSomething()
    {
        try {
            $result = 2;
        } catch (\DummyException $e) {
            throw $e;
        } catch (\NewException $e) {
            throw $e;
        }

        return $result;
    }
}
```

But this passes:
```
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

namespace Magento;

/**
 * Doer
 */
class Doer
{
    /**
     * Do Something
     */
    public function doSomething()
    {
        try {
            $result = 2;
        } catch (\DummyException $e) {
            throw $e;
        }

        return $result;
    }
}
```

Am I missing something?

**Comment by larsroettig** [comment]
@danmooney2 Can you open a bug report than I take a look.
king regards,

Lars

**Comment by danmooney2** [comment]
@larsroettig https://github.com/magento/magento-coding-standard/issues/93

**Comment by larsroettig** [comment]
Thanks, I will take a look this week

**Pull request merged**